src/azure-cli/azure/cli/command_modules/sql/tests/latest/test_sql_commands.py

```diff
@@ -3194,8 +3194,114 @@ def test_sql_managed_db_short_retention(self, resource_group, resource_group_loc
                      self.check('retentionDays', '{retention_days_dec}')])
 
 
-class SqlManagedInstanceRestoreDeletedDbScenarioTest(ScenarioTest):
+class SqlManagedInstanceDbLongTermRetentionScenarioTest(ScenarioTest):
+    @record_only()
+    def test_sql_managed_db_long_term_retention(
+            self):
+
+        self.kwargs.update({
+            'rg': 'clitestxj6awmetud',
+            'loc': 'westus',
+            'managed_instance_name': 'ayang-ltr-mi',
+            'database_name': 'test-1',
+            'weekly_retention': 'P1W',
```

Comment:
I am a little confusing about the allowed values for `weekly_retention`, ...?

Reply:
https://docs.microsoft.com/en-us/azure/sql-database/sql-database-long-term-retention

Here is a little more info.  When using the command line, we expect retention settings to be an ISO 8601 string.

Reply:
Can also explain further on Teams if needed

Reply:
yes. Please ping me in Teams for this.

Reply:
discussed on Teams - users know to use ISO format